code/xcvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -124,15 +124,18 @@ pub struct SolvedOrder {
 impl SolvedOrder {
 	pub fn new(order: OrderItem, solution: Cow) -> StdResult<Self> {
 		ensure!(
-			order.msg.wants.amount.u128() >= solution.given.0,
-			StdError::GenericErr { msg: "user limit was not satisfied".to_string() }
+			order.msg.wants.amount.u128() <= solution.given.u128(),
```

Comment:
Wouldn’t the following work?

```suggestion
			order.msg.wants.amount <= solution.given,
```

Reply:
some time ago it was not working for sure, as they did no impl traits, did not checked now. would it?